src/counts.jl

```diff
@@ -9,53 +9,90 @@
 typealias IntUnitRange{T<:Integer} UnitRange{T}
 
 #### functions for counting a single list of integers (1D)
+"""
+    addcounts!(r, x, levels[, wv])
 
+Fill the array `r` with the number of occurrences in `x` of
+each value in `levels`. If a weighting vector `wv` is specified,
+the sum of weights is used rather than the raw counts.
+"""
 function addcounts!(r::AbstractArray, x::IntegerArray, levels::IntUnitRange)
-	# add counts of integers from x to r
-
-	k = length(levels)
-	length(r) == k || raise_dimerror()
-
-	m0 = levels[1]
-	m1 = levels[end]
-	b = m0 - 1
-
-	@inbounds for i in 1 : length(x)
-		xi = x[i]
-		if m0 <= xi <= m1
-			r[xi - b] += 1
-		end
-	end
-	return r
+    # add counts of integers from x to r
+
+    k = length(levels)
+    length(r) == k || throw(DimensionMismatch())
+
+    m0 = levels[1]
+    m1 = levels[end]
+    b = m0 - 1
+
+    @inbounds for i in 1 : length(x)
+        xi = x[i]
+        if m0 <= xi <= m1
+            r[xi - b] += 1
+        end
+    end
+    return r
 end
 
 function addcounts!(r::AbstractArray, x::IntegerArray, levels::IntUnitRange, wv::WeightVec)
-	k = length(levels)
-	length(r) == k || raise_dimerror()
-
-	m0 = levels[1]
-	m1 = levels[end]
-	b = m0 - 1
-	w = values(wv)
-
-	@inbounds for i in 1 : length(x)
-		xi = x[i]
-		if m0 <= xi <= m1
-			r[xi - b] += w[i]
-		end
-	end
-	return r
+    k = length(levels)
+    length(r) == k || throw(DimensionMismatch())
+
+    m0 = levels[1]
+    m1 = levels[end]
+    b = m0 - 1
+    w = values(wv)
+
+    @inbounds for i in 1 : length(x)
+        xi = x[i]
+        if m0 <= xi <= m1
+            r[xi - b] += w[i]
+        end
+    end
+    return r
 end
 
-counts(x::IntegerArray, levels::IntUnitRange) = addcounts!(zeros(Int, length(levels)), x, levels)
-counts(x::IntegerArray, levels::IntUnitRange, wv::WeightVec) = addcounts!(zeros(eltype(wv), length(levels)), x, levels, wv)
+
+"""
+    counts(x, [levels,] [wv])
+
+Count the number of times that values in the range `levels` occur in
+`x`. The output is a vector of length `length(levels)`. If `levels`
+is not provided, `span(x)` is used in its place. If a weighting
```

Comment:
Yeah, but I think this should be tackled by the help system, e.g. by printing keyword arguments in a different color. That syntax is too useful not to use it.